components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java

```diff
@@ -217,6 +220,20 @@ private void handleUserProvisioning(String username, UserStoreManager userStoreM
         Map<String, String> userClaims = prepareClaimMappings(attributes);
 
         if (userStoreManager.isExistingUser(username)) {
+            String associatedUserName = FrameworkUtils.getFederatedAssociationManager()
+                    .getUserForFederatedAssociation(tenantDomain, idp, subjectVal);
+
+            if (StringUtils.isEmpty(associatedUserName)) {
+                // If a local user is using the same username, association is not allowed unless enabled through config
```

Comment:
Lets end the comment with a fullstop

Reply:
Check other places as well

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5468/commits/90cb4f00ed5abe0a79cebc761f7faa975ee71407